bundles/org.openhab.core.automation.module.script.rulesupport/src/main/java/org/openhab/core/automation/module/script/rulesupport/loader/AbstractScriptFileWatcher.java

```diff
@@ -318,13 +318,16 @@ private synchronized void onStartLevelChanged(int newLevel) {
     }
 
     @Override
-    public void onDependencyChange(@Nullable String scriptPath) {
-        logger.debug("Reimporting {}...", scriptPath);
+    public void onDependencyChange(@Nullable String scriptId) {
+        logger.debug("Reimporting {}...", scriptId);
         try {
-            importFileWhenReady(new ScriptFileReference(new URL(scriptPath)));
-        } catch (MalformedURLException e) {
-            logger.warn("Failed to reimport {} as it cannot be parsed as a URL", scriptPath);
+            ScriptFileReference scriptFileReference = new ScriptFileReference(new URL(scriptId));
+            if (loaded.contains(scriptFileReference)) {
+                importFileWhenReady(scriptFileReference);
+            }
+        } catch (MalformedURLException ignored) {
```

Comment:
It might be handy to have some debug logging here in the future.

Reply:
I decided to ignore it here, because all UI based scripts (or more specifically their id) will throw a `MalformedURLException` because all listeners are notified of all changes, no matter if they really handle this identifier or not. The call is logged by the line below, so I think that this is enough.